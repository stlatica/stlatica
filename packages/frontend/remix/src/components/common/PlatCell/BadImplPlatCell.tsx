import React from "react";
import { MdAutorenew, MdComment, MdStarBorder } from "react-icons/md";

import { PlatMenuButton } from "@/components/button/PlatMenuButton";
import { UserIcon } from "@/components/common/UserIcon";

const dateObj = (dateStr: string): Date => {
  return new Date(dateStr);
};

// const formatter = new Intl.DateTimeFormat("ja-JP", {
//   dateStyle: "medium",
//   timeStyle: "medium",
//   timeZone: "Asia/Tokyo",
// });

// 毎回のフォーマットでインスタンスを生成
const convertDateNotReuseFormatter = (d: Date) => {
  return new Intl.DateTimeFormat("ja-JP", { dateStyle: "medium", timeStyle: "medium" }).format(d);
};

type PlatCellProps = {
  readonly content: string;
  readonly date: string;
  readonly favoriteCount: number;
  readonly replyCount: number;
  readonly shareCount: number;
  readonly userIcon: string;
  readonly userId: string;
  readonly userName: string;
};

/**
 *
 */
export const BadImplPlatCell: React.FC<PlatCellProps> = ({
  content,
  date,
  favoriteCount,
  replyCount,
  shareCount,
  userIcon,
  userId,
  userName,
}) => {
  return (
    <div className="border-2 border-gray-500 bg-gray-800 p-5 text-white">
      {/* ユーザプロファイル */}
      <div className="flex h-12">
        {/* アイコン */}
        <div className="size-12">
          <UserIcon iconImage={userIcon} />
        </div>
        {/* ユーザ名 と ユーザID */}
        <div className="px-3">
          {/* ユーザ名 */}
          <div className="text-lg">{userName}</div>
          {/* ユーザID */}
          <div className="text-sm text-gray-500">{userId}</div>
        </div>
        {/* プルダウンメニュー    参考 : https://mantine.dev/core/menu/ */}
        <div className="ml-auto">
          <PlatMenuButton />
        </div>
      </div>
      {/* platの内容 */}
      <div className="p-5">{content}</div>
      {/* 投稿日時 */}
      <div>
        <div>{convertDateNotReuseFormatter(dateObj(date))}</div>
      </div>
      {/* 下部のアイコン */}
      <div className="flex">
        {/* 返信数 */}
        <div className="flex flex-1 items-center">
          <div className="px-2">
            <MdComment size="2em" />
          </div>
          <div>{replyCount}</div>
        </div>
        {/* シェア数 */}
        <div className="flex flex-1 items-center">
          <div className="px-2">
            <MdAutorenew size="2em" />
          </div>
          <div>{shareCount}</div>
        </div>
        {/* いいね数 */}
        <div className="flex flex-1 items-center">
          <div className="px-2">
            <MdStarBorder size="2em" />
          </div>
          <div>{favoriteCount}</div>
        </div>
      </div>
    </div>
  );
};
