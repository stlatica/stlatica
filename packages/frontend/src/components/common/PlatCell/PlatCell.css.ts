// import { style } from "@vanilla-extract/css";

// export const container = style({
//   border: "2px solid #ccc",
//   backgroundColor: "rgb(31 41 55)",
//   padding: "1.25rem",
//   color: "#fff",
// });

// export const userProfile = style({
//   display: "flex",
//   height: "3rem",
// });

import { style } from "@vanilla-extract/css";

export const container = style({
  border: "2px solid #6b7280", // gray-500
  backgroundColor: "#1f2937", // gray-800
  padding: "1.25rem", // p-5
  color: "#ffffff", // text-white
});

export const userProfile = style({
  display: "flex",
  height: "3rem", // h-12
});

export const userIcon = style({
  width: "3rem", // size-12
  height: "3rem", // size-12
});

export const userDetails = style({
  paddingLeft: "0.75rem", // px-3
});

export const userName = style({
  fontSize: "1.125rem", // text-lg
});

export const userId = style({
  fontSize: "0.875rem", // text-sm
  color: "#6b7280", // text-gray-500
});

export const menuButton = style({
  marginLeft: "auto", // ml-auto
});

export const contentStyle = style({
  padding: "1.25rem", // p-5
});

export const bottomIcons = style({
  display: "flex",
});

export const iconContainer = style({
  display: "flex",
  flex: 1,
  alignItems: "center",
});

export const iconPadding = style({
  paddingLeft: "0.5rem", // px-2
  paddingRight: "0.5rem", // px-2
});
