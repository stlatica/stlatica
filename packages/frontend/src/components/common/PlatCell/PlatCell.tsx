import AutorenewIcon from "@mui/icons-material/Autorenew";
import CommentIcon from "@mui/icons-material/Comment";
import StarBorderIcon from "@mui/icons-material/StarBorder";
import React from "react";

type PlatCellProps = {
  readonly content: string;
  readonly favoriteCount: number;
  readonly replyCount: number;
  readonly shareCount: number;
  readonly userId: string;
  readonly userName: string;
};

/**
 *
 */
export const PlatCell: React.FC<PlatCellProps> = ({
  content,
  favoriteCount,
  replyCount,
  shareCount,
  userId,
  userName,
}) => {
  return (
    <div style={{ padding: "20px", backgroundColor: "#373737", color: "#FFFFFF" }}>
      {/* ユーザプロファイル */}
      <div style={{ display: "flex", height: "50px" }}>
        {/* アイコン */}
        <div
          style={{
            height: "50px",
            width: "50px",
            backgroundColor: "#FFFFFF",
            color: "#000000",
            borderRadius: "100%",
          }}
        />
        <div style={{ paddingLeft: "13px", paddingRight: "13px" }}>
          {/* ユーザ名 */}
          <div style={{ fontSize: "100%", padding: "2%" }}>{userName}</div>
          {/* ユーザID */}
          <div style={{ fontSize: "75%", padding: "2%", color: "#C5C5C5" }}>{userId}</div>
        </div>
      </div>

      {/* platの内容 */}
      <div style={{ padding: "20px" }}>{content}</div>

      {/* 下部のアイコン */}
      <div style={{ display: "flex" }}>
        {/* 返信数 */}
        <div style={{ display: "flex", alignItems: "center", width: "100%" }}>
          <CommentIcon style={{ paddingLeft: "5%", paddingRight: "5%" }} />
          <div>{replyCount}</div>
        </div>
        {/* シェア数 */}
        <div style={{ display: "flex", alignItems: "center", width: "100%" }}>
          <AutorenewIcon style={{ paddingLeft: "5%", paddingRight: "5%" }} />
          <div>{shareCount}</div>
        </div>
        {/* いいね数 */}
        <div style={{ display: "flex", alignItems: "center", width: "100%" }}>
          <StarBorderIcon style={{ paddingLeft: "5%", paddingRight: "5%" }} />
          <div>{favoriteCount}</div>
        </div>
      </div>
    </div>
  );
};
