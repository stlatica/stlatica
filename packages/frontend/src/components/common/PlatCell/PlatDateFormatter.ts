import { MathFloor } from "@/utilities/utilities";

/**
 * 二つの日時データの差を、日・時・分・秒に分解して求める。
 */
export const PlatDateFormatter = (args: { platTimestamp: Date; currentTime: Date }): string => {
  const diff = args.currentTime.getTime() - args.platTimestamp.getTime();

  if (diff <= 0.0) {
    return "現在";
  }

  const seconds = MathFloor(diff / 1000);
  if (seconds < 60) {
    return `${seconds.toString()} 秒前`;
  }

  const minutes = MathFloor(seconds / 60);
  if (minutes < 60) {
    return `${minutes.toString()} 分前`;
  }

  const hours = MathFloor(minutes / 60);
  if (hours < 24) {
    return `${hours.toString()} 時間前`;
  }

  const days = MathFloor(hours / 24);
  if (days < 30) {
    return `${days.toString()} 日前`;
  }

  const months = MathFloor(days / 30);
  if (months < 12) {
    return `${months.toString()} ヶ月前`;
  }

  const years = MathFloor(months / 12);
  return `${years.toString()} 年前`;
};
