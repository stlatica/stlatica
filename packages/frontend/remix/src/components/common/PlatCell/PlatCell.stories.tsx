import { UserIconSampleImage } from "@/components/common/UserIcon/UserIconSampleImage";

import { BadImplPlatCell } from "./BadImplPlatCell";
import { PlatCell } from "./PlatCell";

import type { Story, StoryDefault } from "@ladle/react";

export default {
  title: "common/PlatCell",
} satisfies StoryDefault;

export const Story1: Story = () => {
  return (
    <PlatCell
      content="3週連続でレート下がってるんだが。"
      date="2021-01-30T00:00:00+09:00"
      favoriteCount={4}
      replyCount={0}
      shareCount={1}
      userIcon={UserIconSampleImage}
      userId="@Nobuchi32384"
      userName="夏ノブチ"
    />
  );
};
Story1.storyName = "1行のPlat";

export const Story2: Story = () => {
  return (
    <PlatCell
      content="nobuchiさんのパナソニックグループ プログラミングコンテスト2023（AtCoder Beginner Contest 326）での成績：4526位, パフォーマンス：670相当, レーティング：789→777 (-12) "
      date="2021-01-30T00:00:00+09:00"
      favoriteCount={8}
      replyCount={1}
      shareCount={3}
      userIcon={UserIconSampleImage}
      userId="@Nobuchi79323"
      userName="冬ノブチ"
    />
  );
};
Story2.storyName = "複数行のPlat";

export const Story3: Story = () => {
  return (
    <div className="w-[310px]">
      <PlatCell
        content="nobuchiさんのパナソニックグループ プログラミングコンテスト2023（AtCoder Beginner Contest 326）での成績：4526位, パフォーマンス：670相当, レーティング：789→777 (-12) "
        date="2021-01-30T00:00:00+09:00"
        favoriteCount={8}
        replyCount={1}
        shareCount={3}
        userIcon={UserIconSampleImage}
        userId="@Nobuchi79323"
        userName="冬ノブチ"
      />
    </div>
  );
};
Story3.storyName = "width=310";

export const Story4: Story = () => {
  return (
    <div className="w-[700px]">
      <PlatCell
        content="nobuchiさんのパナソニックグループ プログラミングコンテスト2023（AtCoder Beginner Contest 326）での成績：4526位, パフォーマンス：670相当, レーティング：789→777 (-12) "
        date="2021-01-30T00:00:00+09:00"
        favoriteCount={8}
        replyCount={1}
        shareCount={3}
        userIcon={UserIconSampleImage}
        userId="@Nobuchi79323"
        userName="冬ノブチ"
      />
    </div>
  );
};
Story4.storyName = "width=700";

export const Story5: Story = () => {
  return (
    <div className="w-[1000px]">
      <PlatCell
        content="nobuchiさんのパナソニックグループ プログラミングコンテスト2023（AtCoder Beginner Contest 326）での成績：4526位, パフォーマンス：670相当, レーティング：789→777 (-12) "
        date="2021-01-30T00:00:00+09:00"
        favoriteCount={8}
        replyCount={1}
        shareCount={3}
        userIcon={UserIconSampleImage}
        userId="@Nobuchi79323"
        userName="冬ノブチ"
      />
    </div>
  );
};
Story5.storyName = "width=1000";

export const Story6: Story = () => {
  return (
    <div className="w-[1000px]">
      {Array.from({ length: 3000 }, (_, index) => {
        return (
          <PlatCell
            key={index}
            content="nobuchiさんのパナソニックグループ プログラミングコンテスト2023（AtCoder Beginner Contest 326）での成績：4526位, パフォーマンス：670相当, レーティング：789→777 (-12) "
            date="2021-01-30T00:00:00+09:00"
            favoriteCount={8}
            replyCount={1}
            shareCount={3}
            userIcon={UserIconSampleImage}
            userId="@Nobuchi79323"
            userName="冬ノブチ"
          />
        );
      })}
    </div>
  );
};
Story6.storyName = "DateGoodImpl";

export const Story7: Story = () => {
  return (
    <div className="w-[1000px]">
      {Array.from({ length: 3000 }, (_, index) => {
        return (
          <BadImplPlatCell
            key={index}
            content="nobuchiさんのパナソニックグループ プログラミングコンテスト2023（AtCoder Beginner Contest 326）での成績：4526位, パフォーマンス：670相当, レーティング：789→777 (-12) "
            date="2021-01-30T00:00:00+09:00"
            favoriteCount={8}
            replyCount={1}
            shareCount={3}
            userIcon={UserIconSampleImage}
            userId="@Nobuchi79323"
            userName="冬ノブチ"
          />
        );
      })}
    </div>
  );
};
Story7.storyName = "DateBadImpl";
