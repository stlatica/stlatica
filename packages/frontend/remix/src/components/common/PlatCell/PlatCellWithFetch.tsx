import { UserIconSampleImage } from "@/components/common/UserIcon/UserIconSampleImage";
import { useGetPlat } from "@/openapi/stlaticaInternalApi";

import { PlatCell } from "./PlatCell";

type Props = {
  id: string;
};

export const PlatCellWithFetch: React.FC<Props> = ({ id }) => {
  const { data } = useGetPlat(id, {
    // 認証ヘッダー仮置き
    axios: {
      headers: {
        Authorization: "Bearer aaaa",
      },
    },
  });

  if (data === undefined) {
    return <></>;
  }

  return (
    <PlatCell
      content={data.data.content}
      favoriteCount={20}
      replyCount={1}
      shareCount={10}
      userIcon={UserIconSampleImage}
      userId="test"
      userName="テストユーザー"
    />
  );
};
