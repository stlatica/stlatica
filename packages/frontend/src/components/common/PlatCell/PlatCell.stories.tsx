import React from "react";

import { PlatCell } from "./PlatCell";

import type { Story, StoryDefault } from "@ladle/react";

export default {
  title: "components/common/PlatCell",
} satisfies StoryDefault;

export const Story1: Story = () => {
  return (
    <PlatCell
      content="3週連続でレート下がってるんだが。"
      favoriteCount={4}
      replyCount={0}
      shareCount={1}
      userId="@Nobuchi32384"
      userName="夏ノブチ"
    />
  );
};
Story1.storyName = "1行のPlat";

export const Story2: Story = () => {
  return (
    <PlatCell
      content="nobuchiさんのパナソニックグループ プログラミングコンテスト2023（AtCoder Beginner Contest 326）での成績：4526位, パフォーマンス：670相当, レーティング：789→777 (-12) "
      favoriteCount={8}
      replyCount={1}
      shareCount={3}
      userId="@Nobuchi79323"
      userName="冬ノブチ"
    />
  );
};
Story2.storyName = "複数行のPlat";

export const Story3: Story = () => {
  return (
    <div className="w-[310px]">
      <PlatCell
        content="nobuchiさんのパナソニックグループ プログラミングコンテスト2023（AtCoder Beginner Contest 326）での成績：4526位, パフォーマンス：670相当, レーティング：789→777 (-12) "
        favoriteCount={8}
        replyCount={1}
        shareCount={3}
        userId="@Nobuchi79323"
        userName="冬ノブチ"
      />
    </div>
  );
};
Story3.storyName = "width=310";

export const Story4: Story = () => {
  return (
    <div className="w-[700px]">
      <PlatCell
        content="nobuchiさんのパナソニックグループ プログラミングコンテスト2023（AtCoder Beginner Contest 326）での成績：4526位, パフォーマンス：670相当, レーティング：789→777 (-12) "
        favoriteCount={8}
        replyCount={1}
        shareCount={3}
        userId="@Nobuchi79323"
        userName="冬ノブチ"
      />
    </div>
  );
};
Story4.storyName = "width=700";

export const Story5: Story = () => {
  return (
    <div className="w-1000px]">
      <PlatCell
        content="nobuchiさんのパナソニックグループ プログラミングコンテスト2023（AtCoder Beginner Contest 326）での成績：4526位, パフォーマンス：670相当, レーティング：789→777 (-12) "
        favoriteCount={8}
        replyCount={1}
        shareCount={3}
        userId="@Nobuchi79323"
        userName="冬ノブチ"
      />
    </div>
  );
};
Story5.storyName = "width=1000";
