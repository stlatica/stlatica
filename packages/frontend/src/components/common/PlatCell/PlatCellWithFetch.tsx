import { SampleImageBrown400 } from "@/features/sample/sample-image/SampleImageBrown400";
import { useGetPlat } from "@/openapi/stlaticaInternalApi";

import { PlatCell } from "./PlatCell";

type Props = {
  id: string;
};

export const PlatCellWithFetch: React.FC<Props> = ({ id }) => {
  const { data } = useGetPlat(id, {
    // 認証ヘッダー仮置き
    axios: {
      headers: {
        Authorization: "Bearer aaaa",
      },
    },
  });

  if (data === undefined) {
    return <></>;
  }

  return (
    <PlatCell
      content={data.data.content}
      favoriteCount={20}
      replyCount={1}
      shareCount={10}
      userId="test"
      userName="テストユーザー"
      date={data.data.created_at}
      userIcon={SampleImageBrown400}
    />
  );
};
