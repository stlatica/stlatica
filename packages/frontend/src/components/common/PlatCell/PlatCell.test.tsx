import { render } from "@testing-library/react";
import React from "react";
import { describe, expect, test } from "vitest";

import { Story1, Story2, Story3, Story4, Story5 } from "./PlatCell.stories";

describe("snapshot", () => {
  test("snapshot1", () => {
    const { asFragment } = render(<Story1 />);
    expect(asFragment()).toMatchSnapshot();
  });
  test("snapshot2", () => {
    const { asFragment } = render(<Story2 />);
    expect(asFragment()).toMatchSnapshot();
  });
  test("snapshot3", () => {
    const { asFragment } = render(<Story3 />);
    expect(asFragment()).toMatchSnapshot();
  });
  test("snapshot4", () => {
    const { asFragment } = render(<Story4 />);
    expect(asFragment()).toMatchSnapshot();
  });
  test("snapshot5", () => {
    const { asFragment } = render(<Story5 />);
    expect(asFragment()).toMatchSnapshot();
  });
});
