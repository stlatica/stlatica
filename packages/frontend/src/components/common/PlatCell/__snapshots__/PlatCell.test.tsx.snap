// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`snapshot > snapshot 1`] = `
<DocumentFragment>
  <div
    style="width: 310px; padding: 2%; background-color: rgb(55, 55, 55); color: rgb(255, 255, 255);"
  >
    <div
      style="display: flex; height: 50px;"
    >
      <div
        style="height: 50px; width: 50px; background-color: rgb(255, 255, 255); color: rgb(0, 0, 0); border-radius: 100%;"
      />
      <div
        style="padding-left: 2%; padding-right: 2%;"
      >
        <div
          style="font-size: 100%; padding: 2%;"
        >
          夏ノブチ
        </div>
        <div
          style="font-size: 75%; padding: 2%; color: rgb(197, 197, 197);"
        >
          @Nobuchi32384
        </div>
      </div>
    </div>
    <div
      style="padding: 5%;"
    >
      3週連続でレート下がってるんだが。
    </div>
    <div
      style="display: flex;"
    >
      <div
        style="display: flex; align-items: center; width: 100%;"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="CommentIcon"
          focusable="false"
          style="padding-left: 5%; padding-right: 5%;"
          viewBox="0 0 24 24"
        >
          <path
            d="M21.99 4c0-1.1-.89-2-1.99-2H4c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h14l4 4-.01-18zM18 14H6v-2h12v2zm0-3H6V9h12v2zm0-3H6V6h12v2z"
          />
        </svg>
        <div>
          0
        </div>
      </div>
      <div
        style="display: flex; align-items: center; width: 100%;"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="AutorenewIcon"
          focusable="false"
          style="padding-left: 5%; padding-right: 5%;"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 6v3l4-4-4-4v3c-4.42 0-8 3.58-8 8 0 1.57.46 3.03 1.24 4.26L6.7 14.8c-.45-.83-.7-1.79-.7-2.8 0-3.31 2.69-6 6-6zm6.76 1.74L17.3 9.2c.44.84.7 1.79.7 2.8 0 3.31-2.69 6-6 6v-3l-4 4 4 4v-3c4.42 0 8-3.58 8-8 0-1.57-.46-3.03-1.24-4.26z"
          />
        </svg>
        <div>
          1
        </div>
      </div>
      <div
        style="display: flex; align-items: center; width: 100%;"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="StarBorderIcon"
          focusable="false"
          style="padding-left: 5%; padding-right: 5%;"
          viewBox="0 0 24 24"
        >
          <path
            d="m22 9.24-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
          />
        </svg>
        <div>
          4
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
